lib/prawn/text/formatted/line_wrap.rb

```diff
@@ -101,7 +101,7 @@ def add_fragment_to_line(fragment)
                   @document.width_of(segment, kerning: @kerning)
                 end
 
-              if @accumulated_width + segment_width <= @width
+              if @accumulated_width + segment_width <= @width + 0.00001
```

Comment:
Could you please add a test that demonstrates validity of this fix?

Also could you please explain the value of that number? Why can't it be smaller/bigger?

Reply:
[This commit](https://github.com/unterkoefler/prawn/commit/aca004479435e74664ad348d87253dce96508d01) adds a test that fails, but it does so by manually inserting the necessary rounding errors, so I didn't add it to this pr. Because this bug is so rare, I don't know a good way to add a test for it without changing the code.

Reply:
I chose the value `0.00001` arbitrarily. Seems like it's big enough fix the bug and small enough to not matter visually. The [ruby docs for float](https://ruby-doc.org/3.2.1/Float.html) link to [this page](https://github.com/rdp/ruby_tutorials_core/wiki/Ruby-Talk-FAQ#floats_imprecise) which uses `0.0001` as a delta. Happy to update to that instead if you think that would be better

Reply:
FYI In the geom2d gem which heavily uses floating point calculations I have implemented special comparison methods, see https://github.com/gettalong/geom2d/blob/master/lib/geom2d/utils.rb#L24-L34. The precision to use is arbitrarily set.

For Prawn I would probably choose something that reflects the output precision used in https://github.com/prawnpdf/pdf-core/blob/master/lib/pdf/core/pdf_object.rb#L16

Reply:
Thanks @gettalong! 

@pointlessone I updated to a value based on the precision and also updated one other place doing a similar comparison to use the same value